sql/catalyst/src/main/scala/org/apache/spark/sql/Encoder.scala

```diff
@@ -26,13 +29,51 @@ import org.apache.spark.sql.catalyst.expressions.{DecodeUsingSerializer, BoundRe
 import org.apache.spark.sql.types._
 
 /**
+ * :: Experimental ::
  * Used to convert a JVM object of type `T` to and from the internal Spark SQL representation.
  *
- * Encoders are not intended to be thread-safe and thus they are allow to avoid internal locking
- * and reuse internal buffers to improve performance.
+ * == Scala ==
+ * Encoders are generally created automatically though implicits from a `SQLContext`.
```

Comment:
It would also be great to expand this slightly and explain what can be inferred automatically right now.